cirq/circuits/circuit.py

```diff
@@ -1258,6 +1258,111 @@ def _json_dict_(self):
     def _from_json_dict_(cls, moments, device, **kwargs):
         return cls(moments, strategy=InsertStrategy.EARLIEST, device=device)
 
+    def tetris_concat(
+        *circuits: 'cirq.AbstractCircuit', stop_at_first_alignment: bool = False
```

Comment:
This `stop_at_first_alignment` argument strikes me as very odd. It would seem more natural to me to specify whether the overall alignment should be to push circuits as late or as early as possible, e.g. `align='start'` vs `align='end'`, or something like that. It would seem to me this makes it easier to predict the result without knowing which of two circuits happens to be longer.